lib/storage/file.js

```diff
@@ -620,6 +621,59 @@ File.prototype.delete = function(callback) {
 };
 
 /**
+ * Convenience method to download a file into memory or to a local destination.
+ *
+ * @param {object=} options - Optional configuration. The arguments match those
+ *     passed to {module:storage/file#createReadStream}.
+ * @param {string} options.destination - Local file path to write the file's
+ *     contents to.
+ * @param {function} callback - The callback function.
+ *
+ * @example
+ * //-
+ * // Download a file into memory.
+ * //-
+ * file.download(function(err, contents) {});
+ *
+ * //-
+ * // Download a file to a local destination.
+ * //-
+ * file.download({
+ *   destination: '/Users/stephen/Desktop/file-backup.txt'
+ * }, function(err) {});
+ */
+File.prototype.download = function(options, callback) {
+  if (util.is(options, 'function')) {
+    callback = options;
+    options = {};
+  }
+
+  var destination = options.destination;
+  delete options.destination;
+
+  var fileStream = this.createReadStream(options);
+
+  if (destination) {
+    fileStream
+      .on('error', callback)
+      .pipe(fs.createWriteStream(destination))
+      .on('error', callback)
```

Comment:
That makes sense to me. LMKWYT: https://github.com/GoogleCloudPlatform/gcloud-node/commit/47b69b36e364a015579390f8a8d7e390d73b3b4b